### DISCUSSION TITLE
land_mask flag dictionary in grid-stat using python embedding

### DISCUSSION THREAD

**Discussion by fearon10** [dtcenter/METplus#2059] [Configuration] [answered]
Hi There,

I wanted to inquire how to set the land_mask flag in the config file when running grid-stat. 

The land-sea mask gridded field to be used resides within my forecast file. My forecast file is also read using python embedding. My current logic for using the land_mask flag is pasted below where I have the dictionary set within fcst={} block in the config file. With this logic, I receive output, but the statistics are not stratified by land/ocean. Something is not quite right. In addition, values in my land-ocean grid are 0=water; 1=land; 2=ice, I am not sure how to incorporate these values into the dictionary settings. Ultimately, I am curious if there is working example for the land_mask flag when using python embedding?

Thanks,
Matt

fcst = {
     field = [
        {
         name       = "read_neptune.py -f ci_quarterdeg.hdf5 -m NEPTUNE -l 085000 -lt pressure -v water_vapor_mixing_ratio";
          cat_thresh = [ NA ];
        }
      ];
       land_mask = {
          flag      = TRUE;
          file_name = [PYTHON_NUMPY];
          field     = {name = "read_neptune.py -f ci_quarterdeg.hdf5 -m NEPTUNE -l 10 -lt model_level -v sea_land_ice_mask";}
          regrid    = { method = BILIN; width=2;}
          thresh    = TRUE;
       }
}

**Comment by jprestop** [comment, marked as answer]
Hi @fearon10.  

The use of the land_mask configuration option is for point verification and not for gridded forecasts.  We'd recommend running your forecast file through [gen_vx_mask](https://met.readthedocs.io/en/develop/Users_Guide/masking.html#gen-vx-mask-tool) first to generate a masking region in the form of an output NetCDF mask file.  Then, you should be able to configure the mask.poly entry in the Grid-Stat config file to point to that newly created file.

Please let us know how that goes and if you have any other questions.

**Reply by jprestop** [reply]
Hi @fearon10.  I just wanted to follow up on this Discussion and see if you were able to resolve the problem you were experiencing?  Do you have any remaining questions?

**Reply by fearon10** [reply]
Hi @jprestop. Thank you for your help. For now, I have resolved my issue.

**Reply by jprestop** [reply]
That you @fearon10.  At this point I'm going to lock this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we'll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow.  So if/when more issues/questions arise, please feel free to start a new discussion.
